src/app/editor/components/node-tags-bar/node-tags-bar.component.ts

```diff
@@ -0,0 +1,146 @@
+import { Input, ChangeDetectorRef, ContentChild, ViewChild, Component, OnInit, OnChanges, SimpleChanges } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import { Observable } from 'rxjs/Observable';
+import { DropdownList, InputField, ModalService} from 'gentics-ui-core';
+import { MeshNode } from '../../../common/models/node.model';
+import { fuzzyMatch, fuzzyReplace } from '../../../common/util/fuzzy-search';
+import { ApplicationStateService } from '../../../state/providers/application-state.service';
+import { Tag } from '../../../common/models/tag.model';
+import { EditorEffectsService } from '../../providers/editor-effects.service';
+import { TagReferenceFromServer } from '../../../common/models/server-models';
+import { stringToColor } from '../../../common/util/util';
+import { CreateTagDialogComponent, CreateTagDialogComponentResult } from '../create-tag-dialog/create-tag-dialog.component';
+import { FilterSelection } from '../../../common/models/common.model';
+
+@Component({
+  selector: 'app-node-tags-bar',
+  templateUrl: './node-tags-bar.component.html',
+  styleUrls: ['./node-tags-bar.component.scss']
+})
+export class NodeTagsBarComponent implements OnChanges {
+
+  @ViewChild('DropdownList') dropDown: DropdownList;
+  @ViewChild('InputField') inputField: InputField;
+  @Input() node: MeshNode;
+  isDirty = false;
+  newTagName = ''; // Contains a name for a new tag.
+  nodeTags: TagReferenceFromServer[] = []; //tags for the current opened node
+
+  filteredTags: FilterSelection[] = [];
+
+  constructor(private changeDetector: ChangeDetectorRef,
+    private state: ApplicationStateService,
+    private editorEffects: EditorEffectsService,
+    private sanitized: DomSanitizer,
+    private modalService: ModalService ) { }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.node && changes.node.currentValue) {
+      const node = changes.node.currentValue as MeshNode;
+      this.nodeTags = node.tags ? [...node.tags] : [];
+      this.isDirty = false;
+      this.inputField.writeValue('');
+    }
+  }
+
+  onFilterChange(term: string) {
+
+    this.filterTags(term);
+
+    const tags = Object.values<Tag>(this.state.now.entities.tag);
+    // If the term does NOT perfectly match any of existing tags - we will show an option to create on
+    if (!tags.some(tag => tag.name.toLowerCase() === term.toLowerCase())) {
+        this.newTagName = term;
+    } else {
+        this.newTagName = '';
+    }
+
+    if (!this.dropDown.isOpen) {
+        this.dropDown.openDropdown();
+    }
+    this.dropDown.resize();
+  }
+
+  onTagSelected(tag: Tag) {
+    const { name, tagFamily, uuid } = tag;
+    this.nodeTags.push({ name, tagFamily: tagFamily.name, uuid });
```

Comment:
Let's create a new array here rather than mutating.

This means you can remove the changeDetectorRef.